tasks/task1.test.js

```diff
@@ -0,0 +1,54 @@
+const { range1, range2 } = require("./task1");
+
+// Suite
+describe("Task #1", () => {
+  describe("range1", () => {
+    performTestsFor(range1);
+  });
+  describe("range2", () => {
+    performTestsFor(range2);
+  });
+});
+
+function performTestsFor(func) {
+  test(`should ${func.name}(x=1, y=10) to be [1, 2, 3, 4, 5, 6, 7, 8, 9]`, () => {
+    // Setup
+
+    // Execution
+    const result = func(1, 10);
+    const expected = [1, 2, 3, 4, 5, 6, 7, 8, 9];
+
+    // Assertion
+    expect(result).toEqual(expected);
+  });
```

Comment:
Try to avoid extracting specs to a seperate function. Test should be dumb code:

```
describe("Task #1", () => {
  describe("range1", () => {
    test('...', () => {...});
  });
  describe("range2", () => {
     test('...', () => {...});
  });
});
```

Reply:
In this case, I'd say it's a viable approach to make a handler function to test same logic using both of the functions as nothing differs between tests except the called function. 🤔